hidimstat/desparsified_lasso.py

```diff
@@ -85,68 +139,246 @@ def desparsified_lasso_confint(X, y, confidence=0.95, max_iter=5000,
+    -----------
+    X : ndarray, shape (n_samples, n_features)
+        Data.
+
+    Y : ndarray, shape (n_samples, n_times)
+        Target.
+
+    cov : ndarray, shape (n_times, n_times), optional (default=None)
+        If None, a temporal covariance matrix of the noise is estimated.
+        Otherwise, `cov` is the temporal covariance matrix of the noise.
+
+    test : string, optional (default='chi2')
+        Statistical test used to compute p-values. 'chi2' corresponds
+        to a chi-squared test and 'F' corresponds to an F-test.
+
+    normalize : bool, optional (default=True)
+        If True, the regressors X will be normalized before regression and
+        the target Y will be centered substracting the mean of all the
+        elements of Y to every element of Y.
+
+    max_iter : int, optional (default=5000)
+        The maximum number of iterations when regressing, by Lasso,
+        each column of the design matrix against the others.
+
+    tol : float, optional (default=1e-3)
+        The tolerance for the optimization of the Lasso problems: if the
+        updates are smaller than `tol`, the optimization code checks the
+        dual gap for optimality and continues until it is smaller than `tol`.
+
+    residual_method : string, optional (default='lasso')
+        The method for the computind the residuals of the Nodewise Lasso.
+        Currently the only method available is 'lasso'.
+
+    c : float, optional (default=0.01)
+        Only used if method='lasso'. Then alpha = c * alpha_max.
+
+    noise_method : string, optional (default='simple')
+        If 'simple', the correlation matrix is estimated by taking the
+        median of the correlation between two consecutive time steps
+        and the noise standard deviation for each time step is estimated
+        by taking the median of the standard deviations for every time step.
+        If 'AR', the order of the AR model is given by `order` and the
+        Yule-Walker method is used to estimate the covariance matrix.
+
+    order : int, optional (default=1)
+        If `method=AR`, `order` gives the order of the estimated autoregressive
+        model. `order` must be smaller than the number of time steps.
+
+    n_jobs : int or None, optional (default=1)
+        Number of CPUs to use during the Nodewise Lasso.
+
+    memory : str or joblib.Memory object, optional (default=None)
+        Used to cache the output of the computation of the Nodewise Lasso.
+        By default, no caching is done. If a string is given, it is the path
+        to the caching directory.
+
+    verbose: int, optional (default=1)
+        The verbosity level: if non zero, progress messages are printed
+        when computing the Nodewise Lasso in parralel.
+        The frequency of the messages increases with the verbosity level.
+
+    Returns
+    -------
+    beta_hat : ndarray, shape (n_features, n_times)
+        Estimated parameter matrix.
+
+    sf : ndarray, shape (n_features,)
```

Comment:
why not p-values ?

Reply:
p-values are not signed. We are happy to plot maps with blue or red in the end in general. And in general I always compute sf and cdf rather than p-value for this reason.

Reply:
I think that your terminology is unclear ! Why not signed/unsigned_p_values ?

Reply:
The problem with unsigned_p_values is that you do not know if it is 'survival function' or 'cumulative distribution function' scores. I recall, in the case of a two-sided test:

p-value = 0.05 ---> sf = 0.025 and cdf = 0.975 if effect is positive and sf = 0.975 and cdf = 0.025 if effect is negative.
p-value = 1.00 ---> sf = 0.5 and cdf = 0.5.

And as you probably know, I need to produce the two, to get a better numerical precision (better precision for numbers close to zero than for numbers close to one).

Also sf and cdf are standard quantity. See `scipy.stats.norm.sf` and `scipy.stats.norm.cdf` for example.

Reply:
Ce que je peux faire c'est retourner en plus pval et pval_corr ?

Reply:
left_side_pval and right_side_pval ?